server/events_test.go

```diff
@@ -230,6 +230,15 @@ func TestSystemAccountNewConnection(t *testing.T) {
 	if err := json.Unmarshal(msg.Data, &cem); err != nil {
 		t.Fatalf("Error unmarshalling connect event message: %v", err)
 	}
+	if cem.Schema != "io.nats.server.advisory.v1.client_connect" {
```

Comment:
Make these constants.

Reply:
135 bytes in one of the JS ones, its not a lot :)

Reply:
We could use [URN](https://tools.ietf.org/html/rfc8141)s instead of `io.nats.server.advisory.v1.client_connect` and I think they would be smaller, I'd be ok to switch to that as it's a standard, but I dont think they are widely used.

Reply:
`urn:nats:serverAdvisory:clientConnect` or something like that, close enough to that.

Reply:
How should we go about deciding in your opinion?

Reply:
Difficult question, URNs being a standard carries weight.  They solve the same problem I had which is I want a protocol independant way to point at a schema, can be parsed and transformed into protocol specific ways.

URNs should already have existing parsers in our languages

Reply:
Catching up what was said on the call: I'll read the URN RFC and make sure it can do what we need, if it does we switch.  It's a standard in use by a few popular teams.

Reply:
So I spent a bit of time on this and URNs won't really solve this for us, they identify uniquely the source of event (ie. NATS Server) and the type of event (user connect advisory) but not which version of the event type.  Now we can ignore this restriction and use them anyway - or add a Version field to our structs but see later.

These URNs tend to be used when writing Schemas - ie you write JSON Schema to describe a message - they are not used in the message payloads for association of message to schema.

Cloud Events is closer to what we need as these are also to a large extend events, are also to be consumed by 3rd party systems and address a lot of the issued I had when I first came up with this scheme for us and Choria - and we might even want to support transforming our advisories into Cloud Events in future.

They address this in their spec https://github.com/cloudevents/spec/blob/v1.0/spec.md#type

```
type

    Type: String
    Description: This attribute contains a value describing the type of event related to the originating occurrence. Often this attribute is used for routing, observability, policy enforcement, etc. The format of this is producer defined and might include information such as the version of the type - see Versioning of Attributes in the Primer for more information.
    Constraints:
        REQUIRED
        MUST be a non-empty string
        SHOULD be prefixed with a reverse-DNS name. The prefixed domain dictates the organization which defines the semantics of this event type.
    Examples
        com.github.pull.create
        com.example.object.delete.v2
```

Which is how we got to where we are. So if anything I should change us from `schema` to `type` and this puts us in the best place as already conforming to their `SHOULD` level requirements.

Note on the `io.nats` business which Derek doesnt like, this isn't like Java, this is just the domain part the rest is normal, so we have `io.nats.server.advisory.v1.client_connect` if we did a urn it might be `urn:nats:server.advisory.client_connect`which is no less, its basically `s/io.nats/nats/` and dropping version.

Reply:
https://github.com/nats-io/nats-server/pull/1354/commits/c961ae09ad4b4398987c9047f01dc944d2545298 update to `type` from `schema` here

Reply:
Agree with getting that alignment.